### PR TITLE
update debian packaging

### PR DESCRIPTION
**Pull request by lnussbaum**
Hi,

The changes attached are currently required to build the Debian package properly.

If you are ever interested in getting this into Debian, don't hesite to ask!

Lucas

### CODE REVIEW

File: debian/control [hunk 1]
```diff
@@ -3,7 +3,7 @@ Maintainer: Brenden Blanco <bblanco@plumgrid.com>
 Section: misc
 Priority: optional
 Standards-Version: 3.9.5
-Build-Depends: debhelper (>= 9), cmake, libllvm3.7 | libllvm3.8, llvm-3.7-dev | llvm-3.8-dev, libclang-3.7-dev | libclang-3.8-dev, libelf-dev
+Build-Depends: debhelper (>= 9), cmake, libllvm3.7 | libllvm3.8, llvm-3.7-dev | llvm-3.8-dev, libclang-3.7-dev | libclang-3.8-dev, libelf-dev, bison, flex, libedit-dev, clang-format, python-netaddr, python-pyroute2, luajit, libluajit-5.1-dev
```
Comment (drzaeus77) [inline debian/control]: Note that clang-format, netaddr, and pyroute2 are testing dependencies only; is there a clause to define such a thing? Besides that clang-format also would need clang-format | clang-format-3.7.

**Comment by drzaeus77** [comment]
Adding this to debian would certainly be nice. What are the steps? I'm sure the answer is documented somewhere, but haven't had the time to search it out yet.

**Comment by lnussbaum** [comment]
The steps are: (1) get a working package, conforming to Debian policy; (2) decide who wants to maintain / co-maintain it (I could probably do that); (3) find a sponsor to upload it to Debian (I could do that); (4) enjoy

For (1), the current code sounds like a good basis (but I haven't looked into details yet, my only goal was to get a working package so that I could play with bcc)

**Comment by lnussbaum** [comment]
re testing-only build-deps, yes, there's a way to declare that (mostly to help with architecture bootstrapping) -- see https://wiki.debian.org/BuildProfileSpec . I'll update my patch to add that

File: debian/control (hunk 1 above)
Comment (taem) [inline debian/control]: @drzaeus77 It's build time dependencies only. debhelper has a program to run tests. Binary dependencies listed in the Package sections.

Comment (drzaeus77) [inline debian/control]: Okay that's fine then. The change for `clang-format | clang-format-3.7` is still needed though before we can merge this. It seems that @lnussbaum is busy, I may go ahead and update this myself so we can close the PR.

**Comment by drzaeus77** [comment]
Fixed in other PR with slight modification.

**Comment by vincentbernat** [comment]
Hey! I would also be interested in seeing this in Debian. After a very quick look, the only problem I see in the current packaging is:

 1. The source package needs to be renamed to visor-bcc (bcc already exists in Debian)
 2. The dependencies of the binary packages need to be specified with `${shlib:Depends}` and similar, not hard coded.

@drzaeus77, if you are interested, I can look more carefully and sponsor an upload from you in Debian.

**Comment by lnussbaum** [comment]
Hi @vincentbernat ,

I haven't been able to work on this package, but if you are looking for a co-maintainer, I'm totally willing to help.

Lucas

**Pull request closed without merging**